src/main/resources/templates/timedout.html

```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org">
+<head>
+    <meta charset="UTF-8">
+    <title>Not verified</title>
+</head>
+<body>
+<p th:text="'The link with token ' + ${token} + ' has timed out'">
```

Comment:
I would suggest not to use the "token" word for the final users (many will not understand).
Also I would replace timed out by expired.
So something like: "the link has expired"